app/src/main/java/app/RetrofitClientInstance.kt

```diff
@@ -17,6 +17,6 @@ object RetrofitClientInstance {
                     .addConverterFactory(GsonConverterFactory.create())
                     .build()
             }
-            return retrofit
+            return retrofit!!
```

Comment:
Avoid force unwrap (!!) at all costs.
If the variable contains null, this will throw a NullPointerException and crash the program.
As it were, with Retrofit?, there is not a need for force unwrap.